modules/device/src/main/native/android/dalvik/DalvikDeviceService.java

```diff
@@ -0,0 +1,94 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.helloandroid;
+
+import android.app.Activity;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.provider.Settings.Secure;
+import android.util.Log;
+
+import java.util.UUID;
+
+public class DalvikDeviceService {
+
+    private static final String TAG = "GluonAttach";
+
+    private static final String PREFS_FILE = "device_id.xml";
+    private static final String PREFS_DEVICE_ID = "device_id";
+
+    private final Activity activity;
+    private boolean debug = false;
+
+    public DalvikDeviceService(Activity activity) {
+        this.activity = activity;
+    }
+
+    public void enableDebug() {
+        debug = true;
+    }
+
+    public String getModel() {
+        return Build.MODEL + " ("+ Build.MANUFACTURER + ")";
+    }
+
+    public String getUuid() {
+        final SharedPreferences prefs = activity.getSharedPreferences(PREFS_FILE, 0);
+        String uuid = prefs.getString(PREFS_DEVICE_ID, null);
+        if (uuid != null) {
+            if (debug) {
+                Log.v(TAG, String.format("Retrieved uuid from shared prefs: %s", uuid));
+            }
+            return uuid;
+        }
+
+        final String androidId = Secure.getString(activity.getContentResolver(), Secure.ANDROID_ID);
+        uuid = UUID.nameUUIDFromBytes(androidId.getBytes()).toString();
+        if (debug) {
+            Log.v(TAG, String.format("Storing uuid in shared prefs for future reference: %s", uuid));
+        }
+        prefs.edit().putString(PREFS_DEVICE_ID, uuid).commit();
+        return uuid;
+    }
+
+    public String getPlatform() {
+        return "Android";
+    }
+
+    public String getVersion() {
+        return Build.VERSION.RELEASE + " (" + Build.VERSION.SDK_INT + ")";
+    }
+
+    public String getSerial() {
+        return Build.SERIAL;
```

Comment:
It has been deprecated since API 26. 
It can be replaced with  `Build.getSerial()`, but then requires some kind of privilege permissions, which we don't have.
It will return `unknown` in most of the cases either way, so I guess we can leave it as is for now.

Reply:
Is it used anywhere by other services or any other charm product that you know of? Otherwise it might be a good idea to remove it entirely for now.

Reply:
It i not supported on iOS, and Android returns `unknown`, I'm not sure why do we have it in the first place. +1 on removing it.

Reply:
I have removed it.